src/com/facebook/buck/testrunner/TestNGRunner.java

```diff
@@ -247,4 +289,11 @@ private PrintStream streamToPrintStream(ByteArrayOutputStream str, PrintStream f
       }
     }
   }
+
+  private static class JUnitReportReporterImproved extends JUnitReportReporter {
```

Comment:
`Improved` is not a great term to use. Could you replace it with a name that explain the purpose of this class.